src/index.js

```diff
@@ -280,7 +281,7 @@ export default class GoCommerce {
       }));
     } else {
       return Promise.reject(
-        "Invalid paymentDetails - must have an order_id, an amount and a stripe_token or a paypal_payment_id and paypal_user_id"
+        "Invalid paymentDetails - must have an order_id, an amount, a provider, and a stripe_token or a paypal_payment_id and paypal_user_id"
```

Comment:
Definitely a design decision. I'm on the fence about having the validation on the client-side (beyond the common parameters). With two providers, it is manageable, but I could see it getting complex with more.